lib/Target/TTKernel/TTKernelToCppRegistration.cpp

```diff
@@ -0,0 +1,46 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "mlir/Dialect/EmitC/IR/EmitC.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Tools/mlir-translate/Translation.h"
+
+#include "ttmlir/Dialect/TT/IR/TT.h"
+#include "ttmlir/Dialect/TTKernel/IR/TTKernel.h"
+#include "ttmlir/Dialect/TTKernel/IR/TTKernelOpsTypes.h"
+#include "ttmlir/Target/TTKernel/TTKernelToCpp.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include <mlir/Dialect/MemRef/IR/MemRef.h>
+using namespace mlir;
+
+namespace mlir::tt::ttkernel {
+
+//TODO: Should generalize this to read kernel type from Attribute?
+void registerTensixKernelToCpp() {
+  TranslateFromMLIRRegistration reg(
+      "tensixkernel-to-cpp", "translate tensix kernel to C++",
```

Comment:
Feel like I need to change the option to `ttkernel-to-cpp=noc` and `ttkernel-to-cpp=tensix`

Reply:
Is this possible? If so, where can I find an example?